client/jetpack-cloud/sections/partner-portal/credit-card-fields/credit-card-submit-button.tsx

```diff
@@ -0,0 +1,91 @@
+import { Button, FormStatus, useFormStatus } from '@automattic/composite-checkout';
+import { useElements, CardElement } from '@stripe/react-stripe-js';
+import { useSelect } from '@wordpress/data';
+import { useI18n } from '@wordpress/react-i18n';
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:composite-checkout:credit-card' );
+
+export default function CreditCardSubmitButton( {
+	disabled,
+	onClick,
+	store,
+	stripe,
+	stripeConfiguration,
+	activeButtonText,
+} ) {
+	const { __ } = useI18n();
+	const fields = useSelect( ( select ) => select( 'credit-card' ).getFields() );
+	const useAsPrimaryPaymentMethod = useSelect( ( select ) =>
+		select( 'credit-card' ).useAsPrimaryPaymentMethod()
+	);
+	const cardholderName = fields.cardholderName;
+	const { formStatus } = useFormStatus();
+	const elements = useElements();
+	const cardElement = elements?.getElement( CardElement ) ?? undefined;
+
+	return (
+		<Button
+			// eslint-disable-next-line wpcalypso/jsx-classname-namespace
+			className="button is-primary"
+			disabled={ disabled }
+			onClick={ () => {
+				if ( isCreditCardFormValid( store, __ ) ) {
+					debug( 'submitting stripe payment' );
+					onClick( 'card', {
+						stripe,
+						name: cardholderName?.value,
+						stripeConfiguration,
+						cardElement,
+						useAsPrimaryPaymentMethod,
+					} );
+					return;
+				}
+			} }
+			buttonType="primary"
+			isBusy={ FormStatus.SUBMITTING === formStatus }
+			fullWidth
+		>
+			<ButtonContents formStatus={ formStatus } activeButtonText={ activeButtonText } />
+		</Button>
+	);
+}
+
+function ButtonContents( { formStatus, activeButtonText = undefined } ) {
```

Comment:
Besides making this function a normal one instead of a React component, we can put this entire logic inside of the `CreditCardSubmitButton` function and wrap the content with `useMemo`. That will ensure we are not computing the text if the inputs (formStatus and activeButtonText) have not changed.

The solution would look like: 

```
const buttonContents = useMemo( () => {
	if ( formStatus === FormStatus.SUBMITTING ) {
		return __( 'Processing…' );
	}
	if ( formStatus === FormStatus.READY ) {
		return activeButtonText || __( 'Save payment method' );
	}
	return __( 'Please wait…' );
}, [ formStatus, activeButtonText, __ ] );
```

Reply:
Thanks for this suggestion! Fixed in ad7241beb664069a183a136c802882df5c9335ea